### ISSUE TITLE
Version 3.3 lost my notes and refuses to import previously saved notes

### ISSUE THREAD

**Issue by Cade66** [Sav22999/websites-notes#29]
Up to and including version 3.2.1.1 my notes remained intact.
Starting version 3.3 and until latest 3.3.1.1 : all notes vanished. I tried importing my saved notes (saved with version 3.2.1.1) : refused because "saved under a different version" (or something of the sort).
Reversed to version 3.2.1.1 until an update either keeps my notes either allows me to import the saved ones (from 3.2.1.1).
Annoying. Notefox is quite handy, use it a lot, plenty of notes. I dare not imagine the situation had I not backed up my notes ...

**Comment by Sav22999** [comment]
@Cade66 Hi, thanks for the report.

Although the addon notify you that the version is different, please, press to confirm. The import should work correctly.
I'm so sorry about the inconvenience. I've updated the addon fixing a problem with sync (in fact since 3.3 you'll be able to sync notes with all your devices that use the same Firefox account). From my testings the addon should work correctly from 3.2 to 3.3.
Please, update the addon and try to import notes saved

Let me know, please

**Comment by Sav22999** [comment]
This was caused because I converted all notes from "local" (your computer only) to "sync" (it's synchronized with Firefox account)  – I don't know why it doesn't work for you, I'm so sorry. In future there won't be any issues of this type because I won't change anymore

I'll wait for your news

**Comment by Cade66** [comment]
@Sav22999 I just tried again, unsuccessfully
Firefox 115.1.0 ESR
Running Notefox 3.2.1.1
Updated from about:addons to latest 3.3.1.1
Notes vanished
Tried to import saved notes from 3.2.1.1 : copy/pasted. I even clicked 'Refresh' : nothing.
Version 3.3 manages the 'sync' feature : not sure if this is related but I've disabled this feature : 

```
// disable and hide Firefox Accounts and Sync
pref("identity.fxaccounts.enabled", false);
```

Let me know if you believe disabling 'Sync' could be the culprit, even if disabling it a valid choice included in about:config.
Don't worry, it's not a drama, at least when notes haven't been erased but only inaccessible or backups have been made.

**Comment by Sav22999** [comment]
@Cade66 Ok, please, try to disable that. I'm working to a fix, really really apologise.

**Comment by Cade66** [comment]
> @Cade66 Ok, please, try to disable that. I'm working to a fix, really really apologise.

I did before your reply : pref("identity.fxaccounts.enabled", true); (Default) doesn't fix the issue.
Let's not forget that the issue we're encountering may as always be related to a clash with other extensions nd/or an about:config mofification ...

Don't apologize, you're the one who's doing all the job and issues are often correlated to improvement. You'll find the culprit, but don't forget to sleep ;)

**Comment by Sav22999** [comment]
@Cade66 don't you use a Firefox account?

If you logged in, try to check "Addon" are synced in Account settings ("Settings" > "Sync"), I guess it should be set to on

<img width="687" alt="image" src="https://github.com/Sav22999/websites-notes/assets/32335378/cf520b2b-836c-415a-ad54-27a9e6192fae">

**Comment by Sav22999** [comment]
Can you send me a screenshot of the error the importing process returns? Thanks

**Comment by Cade66** [comment]
I do not use a Firefox account, which is what I meant above with `pref("identity.fxaccounts.enabled", false)` and therefor I have no  ("Settings" > "Sync") in my about:preferences.

It's past one in the morning here so I won't carry on until tomorrow morning.

**Comment by Sav22999** [comment]
@Cade66 Thank you very much. It's night here (01:14 😆 ) but I have to fix this bug before to go to sleep

**Comment by Cade66** [comment]
Here goes : 3 screenshots, sequentially of course. Screen colors are by me (CSS dimming) and last screenshot over-dims by itself ...

![1](https://github.com/Sav22999/websites-notes/assets/65161029/a5e30633-39a0-49e3-bd2b-1f3343cbbc7f)
![2](https://github.com/Sav22999/websites-notes/assets/65161029/649659a1-57f7-42f5-84e0-d64c5f280da5)
![3](https://github.com/Sav22999/websites-notes/assets/65161029/271976b2-1b41-4de7-88e7-9e119b9b4805)


> @Cade66 Thank you very much. It's night here (01:14 😆 ) but I have to fix this bug before to go to sleep

Same time zone here. I know how it is, one can spend a whole night (as i have more than once) to find a/the solution//culprit.
Hope your coffee is good.
See, or rather read you tomorrow morning :)

**Comment by Sav22999** [comment]
Ok, I've just send an update

**Comment by Sav22999** [comment]
@Cade66 The issue should be fixed. In particular:
- If the import doesn't work well (better, now it should work, but maybe for some limitation you cannot use "sync"), you can turn on "save data locally" in settings and it should save as "local" instead of as "sync"
- Fixed issue with importing
- Other improvements

Please, let me know if it works. I apologize again for the inconvenience 😢

**Comment by Cade66** [comment]
@Sav22999 I've just updated Notefox to latest release version 3.3.1.4 and the issue is _**partially**_ resolved.

In `Notefox - Settings`, I set `Save data locally instead of sync with Firefox account` to `True`
I open `Notefox - All Notes` and all notes are in place : OK
I close both Notefox tabs. I reopen `Notefox - All Notes` ... and notes have vanished.
I open again `Notefox - Settings`, verify that `Save data locally instead of sync with Firefox account` is still set to `True` : it is. I enforce by "Save settings" again.
I open again `Notefox - All notes` : no notes
I open again `Notefox - Settings`, I set `Save data locally instead of sync with Firefox account` to `No`, _**then to**_ `Yes`, save settings, open `Notefox - All Notes` and all notes appear correctly 

It seems that the `Save data locally instead of sync with Firefox account` option is not correctly memorized by Notefox 3.3.1.4
I have to reverse to Notefox 3.2.1.1 unfortunately.

**Comment by Sav22999** [comment]
@Cade66 Insert {"storage":"local"}

In this way:

{"notefoox":{...},
 "settings":{...},
 "websites":{...},
...,
 "storage":"local"
}

Do this direclty ok the "import" and let me know. This should work!

**Comment by Cade66** [comment]
@Sav22999 I don't understand : what do you mean by _**Insert {"storage":"local"}**_?
Please elaborate.

**Comment by Sav22999** [comment]
@Cade66 I'm correctly this other issue, let me know some minutes please, and you'll be able to import normally (Settings > Yes, then import) and it should work

**Comment by Sav22999** [comment]
@Cade66 3.3.1.5 published. It's already available on Firefox Addons as well. Please, try now... it should definitely work 🙄

**Comment by Cade66** [comment]
Update to 3.3.1.5
Previous issue `Save data locally instead of sync with Firefox account` not remembered/taken into account remains.
Import resolved, but temporarily. When I re-open Notefox - All Notes ... all notes have vanished.

There are two features,
1- Save data locally instead of sync with Firefox account
2- Import

Notefox 3.3.1.4 resolved feature 1 ... partially (not when re-opening)
Notefox 3.3.1.5 resolves feature 1 & 2 ... partially (not when re-opening).

I don't know if Notefox uses LocalStorage but for what I know an extension is to use IndexedDB, not sessionStorage nor localStorage.
There is obviously IMO a problem in the way Notefox registers settings. But i'm not a coder.

@Sav22999 I won't go on testing at this time. I'm sticking with latest flawless Notefox 3.2.1.1. I'll have a look occasionally to see if updated Notefox versions handle the problem. Good luck. Read you later.

**Comment by Sav22999** [comment]
Last chance... 3.3.1.6 published. If this doesn't work... so, I'm so sorry but I advice you to use 3.2.1.1

**Comment by Sav22999** [comment]
Thank you very much for your time in any case, you'll be precious for me 🙇‍♂️

**Comment by Cade66** [comment]
Updated to 3.3.1.6
Notes are saved for a longer period than before but after opening/closing several tabs, all notes vanish, again.
Anyway, before notes have vanished I still encountered another issue specific to 3.3 I hadn't mentioned given it occurs if all notes are available : the indicator for which pages have notes no longer works. I think this had been reported by a reviewer on Notefox's AMO page as well.

I imagine this is quite a challenge. Unfortunately at this time I have to stick to Notefox 3.2.1.1 in order to have all my notes AND to have the notes indicator work correctly. Right now I have obligations but I remain committed to further testing, later on.

**Comment by Sav22999** [comment]
@Cade66 Thank you for your feedback. I'll do some checks. In the meanwhile I'll disable 3.3+ from Firefox Addons

**Comment by Cade66** [comment]
> @Cade66 Thank you for your feedback. I'll do some checks

The email I just received stated "Thank you for your feedback. I'll do some checks. In the meanwhile I'll disable 3.3+ from Firefox Addons"
Having 3.3+ removed from Firefox AMO would be beneficial for users encountering the same issues as myself when updating extensions, automatically in particular but also "manually" as myself (need to omit Notefox update). I think many users of extensions don't report issues and even less come here to describe them but either remove the problematic extension or rate it poorly on the sole account latest version faces issuues, without any explanation/argument. Hence i believe that removing Notefox 3.3+ from its AMO page would be welcomed at this stage.

**Comment by Sav22999** [comment]
@Cade66 The idea was that, but someone have already updated the add-on, and the 3.3.1.6 works quiete well. I'm checking for the issue you reported and I haven't replicated the bug yet 😕 

If you can describe step-by-step it it would be great to me to fix it

**Comment by Cade66** [comment]
> @Cade66 The idea was that, but someone have already updated the add-on, and the 3.3.1.6 works quiete well. I'm checking for the issue you reported and I haven't replicated the bug yet confused

That someone (who is maybe the one who rated 5 starts on Notefox AMO) may have installed Notefox (likely 3.3.16) as a fresh install.
This led me to the idea of removing my current Notefox and then proceed to a new install : indeed no issue. I didn't even have to set **_Save data locally instead of sync with Firefox account_** to **_Yes_**, everything worked perfectly. Yet, unable to import notes, **even with** **_Save data locally instead of sync with Firefox account_** set to **_Yes_**

My conclusion at this time:
- either proceed to a fresh install of Notefox 3.3.16 and manually try to recover notes 1 by one;
- either stick to Notefox 3.2.1.1
Notefox 3.3.16 fresh install : perfect
Notefox updated to 3.3.16 : the issues mentioned in this lengthy thread

I'll think about it but option 1 means quite a deal of work.

**Comment by Sav22999** [comment]
@Cade66 of course you shouldn't import manually single notes per times, the feature "Import" should work and... I really can't understand why it doesn't work to you because now it works well over all my tests (also with a "synthetic already existing accounts"). I really hate this situation: I've implemented new feature to improve the experience and, instead, it doesn't work correctly. It's a bit frustrating and I'm so sorry

**Comment by Sav22999** [comment]
If you use "sync" it doesn't work yet?

**Comment by Cade66** [comment]
So : Notefox 3.3.16 fresh install works. without importing, default settings, tested a few new notes : no issue.
Import "old" notes does not work unless **Save data locally instead of sync with Firefox account = Yes**
So we set **Save data locally instead of sync with Firefox account = Yes** and Import Notes works, and  the notes indicator work correctly.
Surf a bit, open/close tabs : no issue.
We restart Firefox : notes have vanished, again.

I understand you've tested thoroughly but here I face a problem with no idea about the culprit when:
- updating to 3.3.16 
- installing fresh 3.3.16 and importing previous notes.

I'll have to deal with that. I've tried all possible combinations, sequences and always face the same issues.
If I move to 3.3.16 it'll have to be with a fresh install and re-writing manually all my notes. It's feasible, I have several but not too considerable, perhaps an hour's work.
I think it's no longer needed that we emphasize on this problem. You may close this thread if you wish. Personally I have all the ingredients now to choose lucidly. Thanks.

**Comment by Sav22999** [comment]
@Cade66 Ok, I've understood the issue. When you restart the browser the settings local is reset to "sync", so, for this reason, it looks like you haven't any notes (but actually they are present but the add-on try to keep them from a wrong place). Wait a moment before to import manually, because if the problem is this one, you'll get the same problem ("no notes after restart").

Sorry again

**Comment by Sav22999** [comment]
@Cade66 Notefox 3.3.1.7 published: it should be fixed the start (which the "local"/"sync" where reset automatically to "sync") and icon as well.
Thank you very much for your cooperation and your patience, it's not so common that users help me so much!

**Comment by Cade66** [comment]
One last thing I've just noticed:

- I uninstall my current Notefox
- I proceed to a clean install of Notefox 3.3.16
- I set "Save data locally instead of sync with Firefox account" to "Yes" and I do "Save settings"
- I restart Firefox and I notice that "Save data locally instead of sync with Firefox account" has been reset to "No"

This means that even with a fresh install the process of using Notefox 3.3.16 without the "Sync" feature is problemtic.
Hence, in my experience, Notefox 3.3.16 is totally functional except when

- importing Notes from a pre-3.3 Notefox version
- dealing with the "No Sync" option

I guess 99% of Firefox have not disabled Sync/Firefox account hence no problems.
Seems to me the "no sync" option is not correctly implemented. Sorry to be straightforward but this is how I understand 3.3.16 at this time.
We'll deal with that or not.

**Comment by Sav22999** [comment]
@Cade66 Yes, you're totally true! I've also noticed it (when you told me "when I restart Firefox" I understood the issue: my fault, I implemented this check in a very very bad way... sorry. Now it should work).

Btw I hope you're happy of my support and of my projects: they are all open-source and developed also because I need them too (so, I guess, you can implement better a service if you by self need that service)

**Comment by Cade66** [comment]
> @Cade66 Notefox 3.3.1.7 published: it should be fixed the start (which the "local"/"sync" where reset automatically to "sync") and icon as well. Thank you very much for your cooperation and your patience, it's not so common that users help me so much! (Now you should be able to use "Import" and it should work correctly).

Fresh install of Notefox 3.3.17
"Save data locally instead of sync with Firefox account" : we set to "Yes"
We restart Firefox : "Save data locally instead of sync with Firefox account" is reset to "No", as with Notefox 3.3.16

I'm struggling with this because 1- I love challenges and abandonning is not an option, 2- I really appreciate Notefox, except its new "Sync" by default feature ... because I don't use "Sync", which is not an argument per se.

**Comment by Sav22999** [comment]
@Cade66 ok, but now I can't understand the problem.
I tried many times and it saves correctly the settings... mh. 
Some questions:

1. You press on "Save settings", right?
2. If you don't close Firefox, close "Settings" tab and reopen it, the settings is reset too?

**Comment by Cade66** [comment]
>     1. Have you pressed on "Save settings", right?
>     2. If you don't close Firefox, close "Settings" tab and reopen it, the settings is reset too?

1: YES
2: NO : as long as Firefox is open Notefox 3.3.17 keeps the "Sync" feature (set to "Yes" in my case). The reset to "No" is only after having quit Firefox.

@Sav22999 , you've "tried many times and it saves correctly the settings" : OK. But have you tried with Sync globally disabled for Firefox? :

```
// DISABLE AND HIDE FIREFOX ACCOUNTS AND SYNC
pref("identity.fxaccounts.enabled", false);
```
Last night, with previous 3.3 versions, I had inverted, that is I had tried Notefox 3.3+ with pref("identity.fxaccounts.enabled", false); and that hadn't solved the issues, then. Maybe now it would.

Whatever, the idea is to be able to use Notefox 3.3+ with 'Sync' (and Firefox account) disabled. Disabling "Sync" is after all a user's choice, even ig I presume not many users disable it. It's in about:config so its legitimate to disable it.

**Comment by Sav22999** [comment]
@Cade66 You're right, although about:config permits you also to "broke" Firefox actually. Anyway, I'll try disabling that flag as well.

Have you tried to re-enable for the 3.3.1.7?

Maybe, instead of "reinstall", just "update" the add-on from 3.2.

This is very weird because there is a check when the addon starts (pseudocode) "if exists local variable loads that, otherwise set sync"

The variable can be "sync" or "local" so it should work in any cases, although you disable syncing

**Comment by Sav22999** [comment]
@Cade66 I tried also using the configuration with Firefox account disabled (the flat you post) and it works correctly.
Very weird situation

**Comment by Sav22999** [comment]
I need some other information to try to replicate:
- The OS you are using (Windows, Linux, macOS) and what version of it
- Language of Firefox (so, the language you see in the add-on)
- Version of Firefox (116, 117, ...) and the edition (Firefox, Dev Edition, Beta, Nightly, LTS, ecc.)

Thanks

**Comment by Cade66** [comment]
This time I've updated Notefox to 3.3.1.7, set as before "Save data locally instead of sync with Firefox account" to "Yes", restarted Fitrefox and, surprise, "Yes" remained. I had a look then at "See all notes" and they were all there. Great?

1- The indicator for which pages have notes no longer works
2- "See all notes" suddenly display none, again. Back to settings, "Save data locally instead of sync with Firefox account" and "Yes" is still displayed. I switch to "No" + "Save Settings" and immediately back to "Yes" + "Save Settings' then open "See all notes" and they're all displayed.

I'm giving up. No issue on your side but one step beyond understanding on my side, the Twilight Zone so to say. I'm giving up.
Thanks for your work, for your commitment here but I guess we'll both agree : this issue is more than a brainteaser it's an authentic brain-killer.
Have a nice Sunday evening. I'm off.

Just got notified of your latest comment:

- The OS you are using (Windows, Linux, macOS) and what version of it : **Windows 7 64-BIT**
- Language of Firefox (so, the language you see in the add-on) : **English**
- Version of Firefox (116, 117, ...) and the edition (Firefox, Dev Edition, Beta, Nightly, LTS, ecc.) : **Firefox 115.1.0 ESR**

That'll be all for today as far as i'm concerned. Thanks again.

**Comment by Sav22999** [comment]
@Cade66 Great!
I read some documentation and, theoretically, sync works also if you disable Firefox account (just its behaviour is the same of local). Maybe, I suggest to check if you installed some extensions which causes issues, or maybe you disable other components/settings

This is totally a weird, an authentic brain-killer as you said. Thanks again for your support and have a nice Sunday you too 😍 

I remain at your full disposal

(for now I close this issue, but in case you can reopen it)

**Comment by Sav22999** [comment]
@Cade66 I've sent another update: 3.3.1.8 which disable "sync".
I think the sync in your case doesn't work because there is a space limitation (max 8 KB for all your websites and your notes, which I guess it's too small). I restored the "local" space for everyone and, if you enabled the sync, now the addon show you a message to invite you to disable sync.

I'm sorry for all these issues caused

**Comment by Cade66** [comment]
Bingo! Eureka! & tutti quanti!
Just installed Notefox 3.3.1.8 (clean install, not update), imported notes saved with 3.2.1.1, a warning message stating the notes had been saved with a previous version of Notefox and could be problematic (normal), carried on, all notes restored. Visited pages holding a Notefox note : no problem.

@Sav22999 , nice work and above all nice determination and commitment to find the clue of the enigma. Sincerely: congratulations!
IMO this thread may definitely be closed (considering I opened it): it is now a dead brain-killer, the killer has been spotted and controlled!

**Comment by Sav22999** [comment]
@Cade66 actually thank also to you. Have a nice day!
